### PR TITLE
[Fix] fix coreml for branch 1.x

### PR DESCRIPTION
**Pull request by grimoire**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by Jaykob** [comment]
This is very promising! :-)
Can you give us an example how you would call it for a mmdet model to CoreML?

**Comment by grimoire** [comment]
@lvhan028 https://github.com/open-mmlab/mmdeploy/pull/1658 same fix for branch 1.x

**Review by irexyc — APPROVED** [review]

File: mmdeploy/backend/coreml/torchscript2coreml.py [hunk 1]
```diff
@@ -69,7 +70,7 @@ def from_torchscript(torchscript_model: Union[str,
             default_shape
         convert_to (str, optional): The converted model type, can be
             'neuralnetwork' or 'mlprogram'. Defaults to 'neuralnetwork'.
-        fp16_mode (bool, optional): Convert to fp16 model. Defaults to False.
+        minimum_deployment_target (bool, optional): minimum deploy target.
```
Comment (lvhan028) [inline mmdeploy/backend/coreml/torchscript2coreml.py:73]: bool -> str

File: mmdeploy/backend/coreml/torchscript2coreml.py [hunk 2]
```diff
@@ -52,9 +52,10 @@ def from_torchscript(torchscript_model: Union[str,
                      output_file_prefix: str,
                      input_names: Sequence[str],
                      output_names: Sequence[str],
-                     input_shapes: Dict,
+                     input_shapes: Dict[str, Dict],
+                     compute_precision: str = 'FLOAT32',
                      convert_to: str = 'neuralnetwork',
-                     fp16_mode: bool = False,
+                     minimum_deployment_target: Optional[str] = None,
```
Comment (lvhan028) [inline mmdeploy/backend/coreml/torchscript2coreml.py:58]: What value should be passed to `minimum_deployment_target`?

Comment (grimoire) [reply mmdeploy/backend/coreml/torchscript2coreml.py:58]: According to the [config file](https://github.com/open-mmlab/mmdeploy/blob/master/configs/_base_/backends/coreml.py) I guess it is a string that describes the OS or device. Such as 'IOS16'

Comment (lvhan028) [reply mmdeploy/backend/coreml/torchscript2coreml.py:58]: Let's clarify it more clear in docstring

**Review by lvhan028 — APPROVED** [review]

**Pull request merged**